turtle_graphics/turtle.js

```diff
@@ -0,0 +1,502 @@
+    new Turtle(0, 4)
+        .forward(3)
+        .left()
+        .forward(3)
+        .right()
+        .forward(5)
+        .right()
+        .forward(8)
+        .right()
+        .forward(5)
+        .right()
+        .forward(3)
+        .left()
+        .forward(3)
+        .print();
+    // The above would log the following to the screen:
+    // -- BEGIN LOG
+    // □□□□□□□□□□
+    // □□□■■■■■■□
+    // □□□■□□□□■□
+    // □□□■□□□□■□
+    // ■■■■□□□□■□
+    // □□□□□□□□■□
+    // ■■■■□□□□■□
+    // □□□■□□□□■□
+    // □□□■□□□□■□
+    // □□□■■■■■■□
+    // -- END LOG
+    
+}
+
+//Stretch: As A Script
+//Make the turtle graphics program usable as a script. It should take a string as a an argument that is seperated by dashes (i.e. ->). 
+//This string will contain all turtle commands in abbreviated form:
+// tX,Y for new Turtle where X & Y are numbers representing the starting x & y coordinates. If this command is not given, begin the turtle at (0, 0).
+// fN for forward where N is a number representing how many units the turtle moves forward.
+// r for right
+// l for left
+//Example usage: $ node turtle.js f10-r-r-f10-l-f5-l-f10-r-f5-r-f11
+ // The above would log the following to the screen:
+// •••••••••••
+// •
+// •
+// •
+// •
+// •••••••••••
+//           •
+//           •
+//           •
+//           •
+// •••••••••••
+const turtleCommand = function(arr){
+    //The starting X & Y coordinates
+    let startingX
+    let startingY
+    if (arr[0].includes(",")) {
+        //tX,Y for new Turtle where X & Y are numbers representing the starting x & y coordinates. 
+        startingX = parseInt(arr[0].slice(1).split(",")[0])
+        startingY = parseInt(arr[0].slice(1).split(",")[1])
+        arr.shift()
+    } else {
+        //If this command is not given, begin the turtle at (0, 0).
+        startingX = 0
+        startingY = 0
+    }
+
+    let output = new Turtle(startingX, startingY)
+
+    for (let command of arr) {
+        //fN for forward where N is a number representing how many units the turtle moves forward.
+        if (command[0] === 'f') {
+            output.forward(parseInt(command.slice(1)))
+            //r for right
+        } else if (command === 'r') {
+            output.right()
+            //l for left
+        } else if (command === 'l') {
+            output.left()
+        }
+    }
+    return output.print()
+}
+
+if (process.argv.slice(2).length !== 0 && !process.argv[2].includes('--output=')) {
+    const input = process.argv.slice(2).toString()
+    let arr = input.split("-")
+    turtleCommand(arr)
+}
+
+//Stretch: Save To a File
+//Have your script accept an option --output=[filename<]/code> where [filename] corresponds to the name of a file. 
+//If the option is used, write the turtle drawing to the file using fs.writeFile. Notify the user of that the write was completed.
+
+if (process.argv.slice(2).length !== 0 && process.argv[2].includes('--output=')) {
+    const fs = require('fs');
```

Comment:
in node, common practice is to import modules at the top of the file. this is allowed if you prefer lazy loading but since this is the beginning of bootcamp, I prefer you follow the common practice.